File: cs/Minsk/CodeAnalysis/Diagnostic.cs
namespace Minsk.CodeAnalysis;

public sealed class Diagnostic
{
    internal Diagnostic(TextSpan span, string message)
    {
        Span = span;
        Message = message;
    }

    public TextSpan Span { get; }
    public string Message { get; }

    public override string ToString()
    {
        return Message;
    }
}

File: c/minsk/source/analysis/diagnostic.c
#include "minsk/analysis/diagnostic.h"
#include <stddef.h>

diagnostic_t diagnostic_new(text_span_t span, sds message) {
  return (diagnostic_t){.span = span, .message = message};
}

void diagnostic_free(diagnostic_t *diagnostic) {
  if (diagnostic == NULL || diagnostic->message == NULL) {
    return;
  }
  sdsfree(diagnostic->message);
  diagnostic->message = NULL;
}
